addon/webextension/selector/ui.js

```diff
@@ -198,83 +201,193 @@ window.ui = (function () { // eslint-disable-line no-unused-vars
+               <head>
+                <style>${substitutedCss}</style>
+                <title></title>
+               </head>
+               <body>
+                 <div class="preview-overlay">
+                   <div class="fixed-container">
+                     <div class="preview-instructions"></div>
+                     <div class="myshots-button">
+                       <div class="myshots-text-pre"></div>
+                       <div class="myshots-text"></div>
+                       <div class="myshots-text-post"></div>
+                     </div>
+                     <div class="visible"></div>
+                     <div class="full-page"></div>
+                   </div>
+                 </div>
+               </body>
+              </html>`,
+              "text/html"
+            );
+            this.document = this.element.contentDocument;
+            this.document.replaceChild(
+              this.document.adoptNode(parsedDom.documentElement),
+              this.document.documentElement
+            );
+            installHandlerOnDocument(this.document);
+            if (this.addClassName) {
+              this.document.body.className = this.addClassName;
+            }
+            const overlay = this.document.querySelector(".preview-overlay");
+            overlay.querySelector(".preview-instructions").textContent = browser.i18n.getMessage("screenshotInstructions");
+            overlay.querySelector(".myshots-text").textContent = browser.i18n.getMessage("myShotsLink");
+            overlay.querySelector(".visible").textContent = browser.i18n.getMessage("saveScreenshotVisibleArea");
+            overlay.querySelector(".full-page").textContent = browser.i18n.getMessage("saveScreenshotFullPage");
+            overlay.querySelector(".myshots-button").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onOpenMyShots), false);
+            overlay.querySelector(".visible").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onClickVisible), false);
+            overlay.querySelector(".full-page").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onClickFullPage), false);
+            resolve();
+          });
+          document.body.appendChild(this.element);
+          this.unhide();
+        } else {
+          resolve();
+        }
+      });
     },
 
-    resetPosition: function () {
-      if (! this.movingEl) {
-        return;
+    updateElementSize: function (force) {
+      // Note: if someone sizes down the page, then the iframe will keep the
+      // document from naturally shrinking.  We use force to temporarily hide
+      // the element so that we can tell if the document shrinks
+      const visible = this.element.style.display !== "none";
+      if (force && visible) {
+        this.element.style.display = "none";
+      }
+      let height = Math.max(
+        document.documentElement.clientHeight,
```

Comment:
`updateElementSize` can be simpler for this iframe.  It should always be exactly `window.innerHeight` (and width), and doesn't have to track document changes.  `force` also doesn't matter for this iframe.